tests/posit/connect/external/test_databricks.py

```diff
@@ -0,0 +1,71 @@
+import os
+from typing import Dict
+
+import responses
+from posit.connect import Client
+from posit.connect.external.databricks import (
+    CredentialsProvider,
+    PositCredentialsProvider,
+    PositCredentialsStrategy,
+)
+
+
+def register_mocks():
+    responses.post(
+        "https://connect.example/__api__/v1/oauth/integrations/credentials",
+        match=[
+            responses.matchers.urlencoded_params_matcher(
+                {
+                    "grant_type": "urn:ietf:params:oauth:grant-type:token-exchange",
+                    "subject_token_type": "urn:posit:connect:user-session-token",
+                    "subject_token": "cit",
+                }
+            )
+        ],
+        json={
+            "access_token": "dynamic-viewer-access-token",
+            "issued_token_type": "urn:ietf:params:oauth:token-type:access_token",
+            "token_type": "Bearer",
+        },
+    )
+
+class TestPositCredentialsHelpers:
+    @responses.activate
+    def test_posit_credentials_provider(self):
+        register_mocks()
+
+        client = Client(api_key="12345", url="https://connect.example/")
+        cp = PositCredentialsProvider(posit_oauth=client.oauth, user_session_token="cit")
+        assert cp() == {"Authorization": f"Bearer dynamic-viewer-access-token"}
+
+    @responses.activate
+    def test_posit_credentials_strategy(self):
+        register_mocks()
+
+        class local_strategy:
+            def auth_type(self) -> str:
+                return "local"
+            def __call__(self) -> CredentialsProvider:
+                def inner() -> Dict[str,str]:
+                    return {"Authorization": "Bearer static-pat-token"}
+                return inner
+
+        # local_strategy is used when the content is running locally
+        client = Client(api_key="12345", url="https://connect.example/")
+        cs = PositCredentialsStrategy(local_strategy=local_strategy(),
+                                      user_session_token="cit",
+                                      client=client)
+        cp = cs()
+        assert cs.auth_type() == "local"
+        assert cp() == {"Authorization": "Bearer static-pat-token"}
+
+        # posit_strategy is used when the content is running on Connect
+        os.environ["RSTUDIO_PRODUCT"] = "CONNECT"
```

Comment:
It's possible to mock the environment variables using `unittest.mock.patch`: https://github.com/posit-dev/posit-sdk-py/blob/8648f8038f717514652796634c3804a024642ce2/tests/posit/connect/test_config.py#L7

This avoids modifying the developer's environment entirely.